var/spack/repos/builtin/packages/allinea-forge/package.py

```diff
@@ -26,9 +36,10 @@ class AllineaForge(Package):
 
     def url_for_version(self, version):
         # TODO: add support for other architectures/distributions
+        distro = getDistroName()
         url = "http://content.allinea.com/downloads/"
-        return url + "arm-forge-latest-Redhat-%s-x86_64.tar" % version
-
+        return url + "arm-forge-latest-%s-%s-x86_64.tar" % (distro, version)
```

Comment:
Allinea's version is `latest`! `version` as used here points to the distribution version, and should also be programatically determined.

FWIW, by getting different URLs for the same `version` (in Spack terminology), we're messing with the source hash for mirroring. Not important here, just a future note/warning.